bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BallerinaErrors.java

```diff
@@ -99,15 +100,32 @@ public static ErrorValue createError(BType type, BString message, BString detail
         return new ErrorValue(type, message, null, detailMap);
     }
 
+    public static ErrorValue createDistinctError(String typeIdName, BPackage typeIdPkg, String message) {
+        return createDistinctError(typeIdName, typeIdPkg, message, new MapValueImpl<>(BTypes.typeErrorDetail));
+    }
 
-    public static ErrorValue createDistinctError(String errorCode, String typeIdName, BPackage typeIdPkg,
+    public static ErrorValue createDistinctError(String typeIdName, BPackage typeIdPkg, String message,
                                                  MapValue<BString, Object> detailRecord) {
-        ErrorValue error = createError(errorCode, detailRecord);
+        ErrorValue error = createError(message, detailRecord);
+        setTypeId(typeIdName, typeIdPkg, error);
+        return error;
+    }
+
+    public static ErrorValue createDistinctError(String typeIdName, BPackage typeIdPkg, String message,
+                                                 ErrorValue cause) {
+        MapValueImpl<Object, Object> details = new MapValueImpl<>(BTypes.typeErrorDetail);
+        ErrorValue error = new ErrorValue(new BErrorType(TypeConstants.ERROR, BTypes.typeError.getPackage(),
+                                                         TypeChecker.getType(details)),
+                                          StringUtils.fromString(message), cause, details);
+        setTypeId(typeIdName, typeIdPkg, error);
+        return error;
+    }
```

Comment:
Do we need another constructor taking:
- typeIdName
- pkgId
- message
- cause
- error detail type

Maybe we can add it when there is a requirement for it.

Reply:
+1 We'll add if it is required. Still, we did not find any